Added Palindromic Numbers in Every Language Article (#349)
TheRenegadeCoder/sample-programs-website: projects/palindrome/index.md

---

> title: Palindrome in Every Language

Reviewer: ```suggestion
title: Palindromic Numbers in Every Language
```

layout: default
date: 2020-10-07
last-modified: 2020-10-07
featured-image: 

> tags: [panlindrome]

Reviewer: ```suggestion
tags: [palindrome]
```

authors:
  - anohene1
---

This is an article on how to check if a number is a palindrome in every language.

## Description

> A palindrome is a number that reads the same backward and forward.

Reviewer: I think we should open this up to any string. What do you think? I realize that may mess with your current solution which leverages mod.

Reviewer: The other issue here is the limit in size. Numbers have a limit in most languages.

Author: I will change the Int variables to Long to fix the issue of the limit in size.

Author: I will try and open it up to Strings sometime in the future.

Author: I have finished changing all the Int variables to Long.

Reviewer: Okay! Maybe we can rename the project to Palindromic Number to be specific.

Author: Sure. I'll go ahead and do that

Example: 343, 121, 909, 222

## Requirements

> Create a file called "Even Odd" using the naming convention appropriate for your language of choice.

Reviewer: ```suggestion
Create a file called "Palindrome" using the naming convention appropriate for your language of choice.
```

Author: My bad, I copied that sentence from an already existing file because I wanted everything to be uniform with what is already available and forgot to change it.

Write a program that accepts an input of an integer and prints whether the number is a palindrome or not.

## Testing
The following table contains various test cases that you can use to verify the correctness of your solution:

| Description                  | Input | Output |
|------------------------------|-------|--------|
| no input                     | None  | Usage: please input a number |
| empty input                  | ""    | Usage: please input a number |
| invalid input: not a number  | a     | Usage: please input a number |

> | sample input: palindrome     | 232   | Number is a palindrome       |

Reviewer: I'm wondering if we could simplify this a bit more by outputting true/false directly.
```suggestion
| sample input: palindrome     | 232   | true       |
```

Author: The actual code returns "Number is a palindrome" or "Number is not a palindrome", but I can change it to true or false if you wish.

Reviewer: Yeah, this would make this program consistent with the search algorithms: https://sample-programs.therenegadecoder.com/projects/binary-search/

In general, it'll relieve a lot of headaches related to testing (e.g. someone spells the string wrong)

Author: I have changed them to true or false.

| sample input: not palindrome | 521   | Number is not a palindrome   |

## Articles

> {% include article_list.md collection=site.categories.palindrome % }

Reviewer: ```suggestion
{% include article_list.md collection=site.categories.palindrome %}
```

> {% include article_list.md collection=site.categories.palindrome %}

Reviewer: ```suggestion
{% include article_list.md collection=site.categories.palindromic-numbers %}
```
